znai-core/src/main/java/com/twosigma/znai/extensions/file/CodeReferencesTrait.java

```diff
@@ -42,16 +39,8 @@ public void updateProps(Map<String, Object> props) {
     }
 
     private Map<String, Object> buildReferences() {
-        MarkupTableData tableData = CsvParser.parseWithHeader(
-                componentsRegistry.resourceResolver().textContent(referencesPath),
-                "reference", "url");
-
-        Map<String, Object> result = new LinkedHashMap<>();
-        tableData.forEachRow(row -> {
-            result.put(row.get(0), Collections.singletonMap("pageUrl", row.get(1)));
-        });
-
-        return result;
+        return DocReferencesParser.parse(
+                componentsRegistry.resourceResolver().textContent(referencesPath)).toMap();
```

Comment:
Why don't you use `DocReferencesParser.parse(FileUtils.fileTextContent(referencesFullPath).toMap()`?
I still think that you better avoid the static method to Parse the content.

Reply:
The difference here vs global is here we use a file supplied by a user and we need to find this file according to the way we resolve all files. Hence `resourceResolver`. 
`DocReferencesParser` is designed this way so it can be easily tested and also be used in two different scenarios: user supplied references and global references.